### PR TITLE
Feature/readme support

### PR DESCRIPTION
**Pull request by Reiss-Cashmore**
Added support for readme addon integration and options.

This gives flexibility to users looking to use the block style <docs> readme in SFCs as well as the flexibility of providing a readme property on their stories.

### CODE REVIEW

**Comment by mattrothenberg** [comment]
@Reiss-Cashmore excellent work, I'll get this merged and deployed this weekend

File: src/register.js [hunk 1]
```diff
@@ -37,11 +59,26 @@ export default function registerStories(req, fileName, sbInstance, plugins) {
...
-
+    if(decorators){
+    decorators.forEach((decor) => {
+      storiesOf.addDecorator(decor);
+      });
+    }
     story.knobs ? storiesOf.addDecorator(withKnobs) : false;
 
+    var readmeOptions = storyOptions.readme;
+    var readmeContent = readmeOptions.singleFileComponentBlockEnabled ? componentConfig.default.__docs : componentConfig.default.readme;
```
Comment (mattrothenberg) [inline src/register.js:78]: I'm not sure I understand this bit of code. Also, where does `componentConfig.default.__docs` com from?

File: src/register.js [hunk 2]
```diff
@@ -37,11 +59,26 @@ export default function registerStories(req, fileName, sbInstance, plugins) {
...
+      sidebar: readmeOptions.displaySidebar ? readmeContent : '',
+      content: readmeOptions.displayContent ? readmeContent : '',
+    };
+
     storiesOf.add(story.name, componentFunc, {
-      notes: story.notes
+      notes: story.notes,
+      ...storyOptions,
+      readme: {
+        ...readmeConfiguration
```
Comment (mattrothenberg) [inline src/register.js]: I don't think you need/want to spread this property. Instead, you could do.

```
readme: readmeConfiguration
```

**Comment by mattrothenberg** [comment]
I'm rolling this PR back for the time being, FYI.

**Comment by Reiss-Cashmore** [comment]
No probs, did I break something?

**Comment by mattrothenberg** [comment]
@Reiss-Cashmore Yep. No worries, though!

For now, I think supporting the following APIs is a great place to start.

1. Knobs (which you do an awesome job of implementing)
2. Actions
3. Per-story "options" (e.g., changing the panel position)
4. Decorators (which you did an awesome job of implementing)
5. Notes

**Pull request merged**